File: src/selection.cpp
#include<iostream>
#include "selection.h"
#include "print.h"

using namespace std;

int selection_drive(){
    int n;
    cin >> n;
    int i = 0;
    int arr[n];
    for (int i = 0; i < n; i++) {
        int ele;
        cin >> ele;
        arr[i] = ele;
    }

    selection_sort(arr, n);
    
    print_array(arr, n);

    return 0;
}

void selection_sort(int arr[], int n){
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n - i; j++) {
            if (arr[j+i] < arr[i])
            {
                int temp = arr[j+i];
                arr[j+i] = arr[i];
                arr[i] = temp;
            }
        }
    }
}

File: src/selection.rs
// selection sort
pub fn selection(arr:&mut Vec<usize>, len:usize){
    for i in 0..len {
        for j in 0..len-i {
            if arr.get(i+j).unwrap() < arr.get(i).unwrap() {
                let temp = arr.get(i).unwrap().clone();
                *arr.get_mut(i).unwrap() = *arr.get(i+j).unwrap();
                *arr.get_mut(i+j).unwrap() = temp;
            }
        }
    }
}